cmd/substreams/run.go

```diff
@@ -78,6 +79,16 @@ func runRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("read manifest %q: %w", manifestPath, err)
 	}
 
+	endpoint, err := manifest.ExtractNetworkEndpoint(pkg.Network, mustGetString(cmd, "substreams-endpoint"))
+	if err != nil {
+		fmt.Println(networkDefaultEndpointDeprecationWarning)
```

Comment:
This deprecation warning should live outside of ExtractNetworkEndpoint:
```
substreamsEndpointFlag := mustGetString(cmd, "substreams-endpoint")
if pkg.Network == "" && substreamsEndpointFlag == "" {
 // trigger deprecation warning logic
} else {
// manifest.ExtractNetworkEndpoint(...)
}
```

Reply:
This way the error is returned for every other case

Reply:
Also, there should be a CHANGELOG entry and README entry about this:
"here's how you can easily handle different networks in substreams: set your endpoints to each environment variable, ....." etc.